servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -647,13 +672,12 @@ public void heartbeat() {
             LOG.error("The worker " + worker + " got timed out!");
             mLostWorkers.add(worker);
             iter.remove();
-          } else if (mLostWorkers.contains(worker)) {
-            LOG.info("The lost worker " + worker + " is found.");
-            mLostWorkers.remove(worker);
```

Comment:
Could you explain why removing this? If a worker is detected again, `mLostWorkers` and the lost files should be updated accordingly, right?